uni/lib/view/common_widgets/pages_layouts/general/general.dart

```diff
@@ -15,13 +15,15 @@ import 'package:uni/view/common_widgets/pages_layouts/general/widgets/top_naviga
 abstract class GeneralPageViewState<T extends StatefulWidget> extends State<T> {
   bool _loadedOnce = false;
   bool _loading = true;
+  Future<void> Function(BuildContext)? onLoad;
 
   Future<void> onRefresh(BuildContext context);
 
-  Future<void> onLoad(BuildContext context) async {}
-
   @override
   Widget build(BuildContext context) {
+    if (onLoad == null) {
```

Comment:
It think it is cleaner to simply surround the callback with the condition, this is repeated below.

Reply:
If the callback is surrounded by `if (onLoad != null) {...}` the variable `_loading` should be also set to false so the condition `else { _loading = false; }` is needed. Do you still prefer this approach over the implemented one?

Reply:
Hm, I meant surround the add callback statement, not the callback body

Reply:
But still, I get that, can we set loading and onLoad not null?

Reply:
Yep, it is feasible, but I think that in this case, the return early pattern makes the code simpler and easier to read

Reply:
Yeah, seems like a tradeoff. In that case, I'll leave it to your preference.